pkg/machine/fcos.go

```diff
@@ -91,24 +90,19 @@ func UpdateAvailable(d *Download) (bool, error) {
 	//	 check the sha of the local image if it exists
 	//  get the sha of the remote image
 	// == dont bother to pull
-	files, err := ioutil.ReadDir(filepath.Dir(d.LocalPath))
+	if _, err := os.Stat(d.LocalPath); os.IsNotExist(err) {
+		return false, nil
+	}
+	fd, err := os.Open(d.LocalPath)
 	if err != nil {
 		return false, err
 	}
-	for _, file := range files {
-		if filepath.Base(d.LocalPath) == file.Name() {
-			b, err := ioutil.ReadFile(d.LocalPath)
-			if err != nil {
-				return false, err
-			}
-			s := sha256.Sum256(b)
-			sum := digest.NewDigestFromBytes(digest.SHA256, s[:])
-			if sum.Encoded() == d.Sha256sum {
-				return true, nil
-			}
-		}
+	defer fd.Close()
```

Comment:
can you catch the error and use logrus to dump it?

Reply:
yes done